rust_snuba/src/processors/metrics_summaries.rs

```diff
@@ -34,27 +44,39 @@ pub fn process_message(
                 max: summary.max,
                 metric_mri,
                 min: summary.min,
-                project_id: from.project_id,
-                retention_days: enforce_retention(from.retention_days, &config.env_config),
-                span_id: from.span_id,
+                project_id: span.project_id,
+                retention_days,
+                span_id: span.span_id,
                 sum: summary.sum,
                 tag_keys,
                 tag_values,
-                trace_id: from.trace_id,
+                trace_id: span.trace_id,
             })
         }
     }
 
-    InsertBatch::from_rows(metrics_summaries)
+    let origin_timestamp = DateTime::from_timestamp(span.received as i64, 0);
+
+    Ok(InsertBatch {
+        rows: RowData::from_rows(metrics_summaries)?,
+        origin_timestamp,
+        sentry_received_timestamp: None,
+    })
 }
 
 #[derive(Debug, Default, Deserialize)]
 struct InputMessage {
-    #[serde(default)]
-    _metrics_summary: BTreeMap<String, Vec<FromMetricsSummary>>,
+    _metrics_summary: Option<Box<RawValue>>,
```

Comment:
I believe [IgnoredAny](https://docs.rs/serde/latest/serde/de/struct.IgnoredAny.html) might fit your purpose even more narrowly -- it isn't specific to `serde-json`, and AFAIK there's fewer hacks in there because it doesn't allow you to access the inner value as a string and really just discards. who knows if it is faster though